### ISSUE TITLE
"or_fun_call" lint not triggered for method calls

### ISSUE THREAD

**Issue by shepmaster** [Manishearth/rust-clippy#1415]
```rust 
fn main() {
    // This is fine
    None.unwrap_or(""); 
    // This is non-ideal, but warned about, yay!
    None.unwrap_or(String::from("")); 
    // This is equally non-ideal, but not warned about, boo!
    None.unwrap_or("".to_owned()); 
}
```

Very tangentially related to #1090.

- Clippy 0.0.104
- rustc 1.15.0-nightly (71c06a56a 2016-12-18)

**Comment by llogiq** [comment]
Apart from `&str.`{`to_owned`, `to_string`}`()`, are there other methods we should lint?
